### PR TITLE
Market Watcher stability fix - Closes #545

### LINKED ISSUE #545: Market Watcher and Price Ticker stability bugs
This error can lead to random app crashes.
Observed only on a local machine so far.

```
TypeError: Cannot convert undefined or null to object
    at BittrexOrders.__dirname.BittrexOrders.rest.acceptOrders (/Users/michal/work/lisk-explorer/lib/orders/bittrex.js:62:26)
    at Request.__dirname.BittrexOrders.rest.retrieveOrders.request.get [as _callback] (/Users/michal/work/lisk-explorer/lib/orders/bittrex.js:52:26)
    at Request.init.self.callback (/Users/michal/work/lisk-explorer/node_modules/request/request.js:186:22)
    at emitTwo (events.js:106:13)
    at Request.emit (events.js:191:7)
    at Request.<anonymous> (/Users/michal/work/lisk-explorer/node_modules/request/request.js:1163:10)
    at emitOne (events.js:96:13)
    at Request.emit (events.js:188:7)
    at IncomingMessage.<anonymous> (/Users/michal/work/lisk-explorer/node_modules/request/request.js:1085:12)
    at IncomingMessage.g (events.js:292:16)
```

### PR DESCRIPTION
**Pull request by MichalTuleja**
### What was the problem?
There are random errors during Market Watcher updates that lead to process crashes.

### How did I fix it?
Additional validation logic has been added to the parts of the code that depends on 3rd party services.

### How to test it?
The app should be stable on devnet/testnet/betanet.

### Review checklist
- The PR solves #545
- All new code is covered with unit tests
- All new features are covered with e2e tests
- All new code follows best practices

### CODE REVIEW

File: lib/orders/abstract.js [hunk 1]
```diff
@@ -48,10 +48,16 @@ function AbstractOrders(client) {
...
+				if (!Array.isArray(result.asks)) return false;
+				if (!Array.isArray(result.bids)) return false;
+				return true;
+			};
+
 			const message = body[self.response.error];
 			if (message) {
 				return cb(message);
-			} else if (_.isObject(body)) {
+			} else if (_.isObject(body) && validateResult(body)) {
```
Comment (lsilvs) [inline lib/orders/abstract.js]: I would add `_.isObject(body)` check inside validateResult function

File: lib/orders/bittrex.js [hunk 2]
```diff
@@ -45,10 +45,16 @@ function BittrexOrders(...rest) {
...
+				if (!Array.isArray(result.buy)) return false;
+				if (!Array.isArray(result.sell)) return false;
+				return true;
+			};
+
 			const message = body.message;
 			if (message) {
 				return cb(message);
-			} else if (_.isObject(body) && _.isObject(body.result)) {
+			} else if (_.isObject(body) && _.isObject(body.result) && validateResult(body.result)) {
```
Comment (lsilvs) [inline lib/orders/bittrex.js]: Same thing here

File: utils/exchange-api/index.js [hunk 3]
```diff
@@ -70,7 +78,7 @@ module.exports = function (config) {
 					if (res.error) {
 						return cb(res.error);
 					}
-					return cb(null, res.BTC_RUB.last_trade);
+					return cb(null, safeRef(res, 'BTC_RUB.last_trade'));
```
Comment (lsilvs) [inline utils/exchange-api/index.js]: Does this work?

File: utils/exchange-api/index.js [hunk 4]
```diff
@@ -82,7 +90,7 @@ module.exports = function (config) {
 					if (res.error) {
 						return cb(res.error);
 					}
-					return cb(null, res.BTC_LSK.last);
+					return cb(null, safeRef(res, 'BTC_LSK.last'));
```
Comment (lsilvs) [inline utils/exchange-api/index.js]: Does this work?

File: utils/exchange-api/index.js (hunk 3 above)
Comment (MichalTuleja) [reply utils/exchange-api/index.js]: Should work now.

File: utils/exchange-api/index.js (hunk 4 above)
Comment (MichalTuleja) [reply utils/exchange-api/index.js]: Should work now.

**Pull request merged**